Lib/pdb.py

```diff
@@ -622,11 +624,97 @@ def _disable_command_completion(self):
             self.completenames = completenames
         return
 
+    def _exec_in_closure(self, source, globals, locals):
+        """ Run source code in closure so code object created within source
+            can find variables in locals correctly
+
+            returns True if the source is executed, False otherwise
+        """
+
+        # Determine if the source should be executed in closure. Only when the
+        # source compiled to multiple code objects, we should use this feature.
+        # Otherwise, we can just raise an exception and normal exec will be used.
+
+        code = compile(source, "<string>", "exec")
+        for const in code.co_consts:
+            if isinstance(const, CodeType):
+                break
+        else:
+            return False
```

Comment:
Nice simplification to check for code objects here.

I find this a little clearer than `for`/`break`/`else`:

```suggestion
        if any(isinstance(const, CodeType) for const in code.co_consts):
            return False
```